cmd/tikv-server/src/main.rs

```diff
@@ -210,8 +225,84 @@ fn main() {
         process::exit(0);
     }
 
+    // This is for creating a shadow instance from exists data, which is specified
+    // by `config.data_dir`. When duplicating all data files, *symlink* will be used
+    // if possible, otherwise *copy* will be used instead.
+    if let Some(agent_dir) = matches.value_of("create-fork") {
+        if data_key_manager_from_config(&config.security.encryption, &config.storage.data_dir)
+            .unwrap()
+            .is_some()
+        {
+            eprintln!("fork with encryption enabled is not expected");
+            process::exit(-1);
+        }
+
+        if let Err(e) = remove_and_recreate_dir(agent_dir) {
```

Comment:
We should probably make sure the directory doesn't exist instead. Otherwise it's easy to delete the directory mistakenly.